tests/test_scenarios.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env python
+import logging
+from pathlib import Path
+
+import pytest
+from conftest import TEST_Path
+from pyrdfj2 import J2RDFSyntaxBuilder
+from util4tests import run_single_test
+
+from travharv import TravHarv
+
+log = logging.getLogger(__name__)
+
+QUERY_BUILDER: J2RDFSyntaxBuilder = J2RDFSyntaxBuilder(
+    templates_folder=str(Path(__file__).parent / "scenarios" / "templates")
+)
+
+BASE = "http://localhost:8080/"
+OUTPUTS = TEST_Path / "output"
+INPUT = TEST_Path / "input"
+CONFIGS = TEST_Path / "travharv_config"
+RDF_MIMES = {
+    "text/turtle",
+    "application/ld+json",
+}
+
+# TODO write out all the assertions for the tests
+# this being the number of triples
+# the documents that ware asserted
+SCENARIOS_OUTCOMES = {
+    "dereference_test1_sparql.yml": [
+        {"expected_len_triples": "19", "mode": "stop"},
+        {"expected_len_triples": "19", "mode": "nostop"},
+    ],
+    "dereference_test2_sparql.yml": [
+        {"expected_len_triples": "28", "mode": "stop"},
+        {"expected_len_triples": "TBD", "mode": "nostop"},
+    ],
+    "dereference_test3_sparql.yml": [
+        {"expected_len_triples": "19", "mode": "stop"},
+        {"expected_len_triples": "TBD", "mode": "nostop"},
+    ],
+    "dereference_test4_sparql.yml": [
+        {"expected_len_triples": "19", "mode": "stop"},
+        {"expected_len_triples": "TBD", "mode": "nostop"},
+    ],
+    "dereference_test5_sparql.yml": [
+        {"expected_len_triples": "19", "mode": "stop"},
+        {"expected_len_triples": "TBD", "mode": "nostop"},
+    ],
+}
+
+
+# launch a server with subprocess from local_server.py
+@pytest.mark.usefixtures("httpd_server_base", "store_info_sets")
+def test_scenarios(
+    httpd_server_base: str,
+    store_info_sets,
+):
+    assert httpd_server_base
+
+    for config in CONFIGS.glob("*.yml"):
+        if config.name not in SCENARIOS_OUTCOMES:
+            log.debug(f"Skipping scenario {config}")
+            continue
+        for test_case in SCENARIOS_OUTCOMES[config.name]:
```

Comment:
strange construct -- why not simply nest looping the config-names from the dict???
very confusing approach to glob and then ignore - see no reason to do that?

Reply:
to select the specific tests I want to run. If i have a test that keeps failing I don't want to wait for the other ones. And I'm not going to write a seperate test to to run1 scenario

Reply:
still does not click with me
looks like you could simply loop the config_names at the top-level of SCENARIO_OUTCOMES ?

those should be carefully selected to actually match names that exist in the CONFIGS folder anyway?